### DISCUSSION TITLE
Does this project consider support null-ls?

### DISCUSSION THREAD

**Discussion by cathaysia** [williamboman/nvim-lsp-installer#483] [Q&A] [answered]
null-ls provide some lsps, and it need external tool also.  such as https://github.com/cheshirekow/cmake_format provide formatter for CMakeLists.txt. Other than cmake-language-server, it brings more setting to cmake.

I want add a installer for https://github.com/cheshirekow/cmake_format, but I don't know whether that can be merged.

**Comment by williamboman** [comment, marked as answer]
Hello and yes! However, I'd like to make some core/fundamental changes to certain things before opening up for things other than LSP - a big one being to not interfere with other plugins by forcing users to call into nvim-lsp-installer APIs to set things up (i.e. getting rid of `lsp_installer.on_server_ready()`).

**Comment by cathaysia** [comment]
Thanks. 🙂

In my opinion, we can give all tools a suffix to distinguish them from LSPs, such as cmake_format_tool, or clang_tidy_tool. In this way, should just a little codes needed changed.

I had wrote some codes to do this, but it has no test:

![图片](https://user-images.githubusercontent.com/42114817/154183943-3b633734-f446-4038-8e51-d176d7d4d71f.png)

```lua
function M.on_server_ready(cb)
    dispatcher.register_server_ready_callback(cb)
    vim.schedule(function()
        local installed_servers = servers.get_installed_servers()
        for i = 1, #installed_servers do
            local endsWith = function (str, substr)
                if str == nil or substr == nil then
                    return nil
                end
                if str:reverse():find(substr:reverse()) ~= 1 then
                    return false
                end
                return true
            end
            if not endsWith(installed_servers[i].name, 'tool') then
                dispatcher.dispatch_server_ready(installed_servers[i])
            end
        end
    end)
end
```
